### PR TITLE
DM-27663: Update ts_observatory_control other packages to stop using Rotator application topic

### PR DESCRIPTION
**Pull request by tribeiro**
Remove usage of deprecated rotator application topic in favor of rotation.
Remove pre-commit hook and add pre-commit configuration.
Add pytest-back.

### CODE REVIEW

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 1]
```diff
@@ -313,11 +313,15 @@ async def monitor_position(self, check=None):
                 )
 
             if _check.mtrotator:
-                rotator = await self.rem.mtrotator.tel_application.next(
+                rotator = await self.rem.mtrotator.tel_rotation.next(
```
Comment (r-owen) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:316]: Please consider `rotation` or `rotation_data` for the data. It is a better match to the telemetry topic name.

(In a unit test you use a completely different variable name for the same thing, which I complain about below).

**Review by r-owen — CHANGES_REQUESTED** [review]
I think I found one bug. Otherwise looks good. Thanks for making this change.

File: python/lsst/ts/observatory/control/mock/mtcs_mock.py [hunk 2]
```diff
@@ -199,8 +199,8 @@ async def rotator_telemetry(self):
             ):
 
                 error = np.random.normal(0.0, 1e-7)
-                demand = self.controllers.mtrotator.tel_application.data.demand
-                position = self.controllers.mtrotator.tel_application.data.position
+                demand = self.controllers.mtrotator.tel_rotation.data.demandPosition
+                position = self.controllers.mtrotator.tel_rotation.data.actualPosition
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/mock/mtcs_mock.py:203]: Sorry, not sure I understood this comment. Is it possible you are confusing `Controller` with `Remote`? `self.controllers.mtrotator` is a controller for `MTRotator`, this 

```
self.controllers.mtrotator.tel_rotation.data
```

will never return `None` right?

**Comment by tribeiro** [comment]
I think I implemented all your comments, except for the one which I didn't understand.

File: python/lsst/ts/observatory/control/mock/mtcs_mock.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (r-owen) [reply python/lsst/ts/observatory/control/mock/mtcs_mock.py:204]: You are right; I had missed that. A WriteTopic will always return a sample.

However, that sample will contain initialized values (which may not be realistic) until you have written it once. Is that safe here? If not then it's still worth some protection. A code comment would be appreciated in any case.

Note that WriteTopic has a property `has_data` which allows you to test if you have ever written data. So you could modify my incorrect suggestion with:
```
if not self.controllers.mtrotator.tel_rotation.has_data:
    # return or do something else intelligent
rotation = self.controllers.mtrotator.tel_rotation.data
rot_delta = ...
```
I suppose the fact that no data has been written will result in rot_delta = 0, which _might_ be fine. But I hope you will think about what seems best and add a code comment explaining.

**Review by r-owen — APPROVED** [review]
I replied to your one comment. I am happy to have you merge it once you have thought about that one a bit more.

**Pull request merged**